### PR TITLE
ci: use `h3-nightly@latest` for edge releases

### PR DESCRIPTION
**Pull request by pi0**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: scripts/bump-edge.ts [hunk 1]
```diff
@@ -117,6 +119,11 @@ async function main() {
       `${pkg.data.version}-${date}.${commit}`
     );
     workspace.rename(pkg.data.name, pkg.data.name + "-edge");
+    pkg.updateDeps((dep) => {
+      if (nightlyPackages[dep.name]) {
+        dep.range = "npm:" + nightlyPackages[dep.name] + "@latest";
```
Comment (danielroe) [inline scripts/bump-edge.ts:124]: see my comment at https://github.com/nuxt/nuxt/pull/22593#discussion_r1291241882.

I'm not sure `latest` is safe here. I'd recommend a caret + version hash.

Comment (pi0) [reply scripts/bump-edge.ts:124]: Also see my comment there! I think it should be safer than pinned version to avoid hoisting issues on edge.

Comment (pi0) [reply scripts/bump-edge.ts:124]: Reposting from https://github.com/nuxt/nuxt/pull/22593#discussion_r1291262322, caret + hash is respected by package manager in a way that it won't look for any newer version + (sadly) `h3-nightly@^1` doesn't works with packages published with `-` version.

**Comment by pi0** [comment]
Merging to test end-to-end with the package manager behavior + coupled with nuxt edge.

I think if it is introducing to much trouble, we shall revert it and keep using `h3` latest stable release. 1.8 was kinda exceptional release that taking too much rc(s).

**Pull request merged**